src/tests/Interop/PInvoke/Decimal/DecimalTest.cs

```diff
@@ -4,32 +4,44 @@
 using System;
 using System.Runtime.InteropServices;
 using TestLibrary;
+using FactAttribute = Xunit.FactAttribute;
 
-public class DecimalTest
+namespace Xunit
 {
-    private const int StartingIntValue = 42;
-    private const int NewIntValue = 18;
+    [AttributeUsage(AttributeTargets.Assembly | AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = true, Inherited = true)]
+    public class SkipOnPlatformAttribute : Attribute
+    {
+        internal SkipOnPlatformAttribute() { }
+        public SkipOnPlatformAttribute(TestPlatforms testPlatforms, string reason) { }
+    }
 
-    public static int Main()
+    [Flags]
+    public enum TestPlatforms
```

Comment:
Hmm, I hope we won't need to define the platform enumeration in every single test getting converted to the new model, will we? I would assume this should come from some common reference module, ideally shared with the Roslyn source generator. I noticed a discussion about <code>TestPlatforms</code> in the source generator code itself, I'm not sure to what extent these two can be reused either by compiling the same source file twice or by using some common dependent assembly but duplicating the enumeration in test code clearly sucks.